tests/TestCase/View/Helper/FormHelperTest.php

```diff
@@ -8496,4 +8498,266 @@ public function testNestedLabelInput()
+            'My checkboxes',
+            '/label',
+            'input' => ['type' => 'hidden', 'name' => 'checkbox1', 'value' => ''],
+            ['div' => ['class' => 'checkbox']],
+            ['input' => ['type' => 'checkbox', 'name' => 'checkbox1[]', 'value' => '1', 'id' => 'checkbox1-1']],
+            '/div',
+            ['div' => ['class' => 'checkbox']],
+            ['input' => ['type' => 'checkbox', 'name' => 'checkbox1[]', 'value' => '2', 'id' => 'checkbox1-2']],
+            '/div',
+            '/div'
+        ];
+        $this->assertHtml($expected, $result);
+
+        // Tests class as array, adding selected to class with custom attributes (MultiCheckbox)
+        $result = $this->Form->input('checkbox1', [
+            'label' => 'My checkboxes',
+            'multiple' => 'checkbox',
+            'type' => 'select',
+            'options' => [
+                ['text' => 'First Checkbox', 'value' => 1],
+                ['text' => 'Second Checkbox', 'value' => 2]
+            ],
+            'optionsLabel' => ['class' => 'custom-class'],
+            'value' => ['1']
+        ]);
+
+        $expected = [
+            ['div' => ['class' => 'input select']],
+            ['label' => ['for' => 'checkbox1']],
+            'My checkboxes',
+            '/label',
+            'input' => ['type' => 'hidden', 'name' => 'checkbox1', 'value' => ''],
+            ['div' => ['class' => 'checkbox']],
+            ['label' => [
+                'class' => 'custom-class selected',
+                'for' => 'checkbox1-1'
+            ]],
+            ['input' => [
+                'type' => 'checkbox',
+                'name' => 'checkbox1[]',
+                'value' => '1',
+                'id' => 'checkbox1-1',
+                'checked' => 'checked'
+            ]],
+            'First Checkbox',
+            '/label',
+            '/div',
+            ['div' => ['class' => 'checkbox']],
+            ['label' => [
+                'class' => 'custom-class',
+                'for' => 'checkbox1-2'
+            ]],
+            ['input' => [
+                'type' => 'checkbox',
+                'name' => 'checkbox1[]',
+                'value' => '2',
+                'id' => 'checkbox1-2'
+            ]],
+            'Second Checkbox',
+            '/label',
+            '/div',
+            '/div'
+        ];
+        $this->assertHtml($expected, $result);
+
+        // Tests class as array, adding selected to class with custom attributes (MultiCheckbox)
+        $result = $this->Form->input('checkbox1', [
+            'label' => 'My checkboxes',
+            'multiple' => 'checkbox',
+            'type' => 'select',
+            'options' => [
+                ['text' => 'First Checkbox', 'value' => 1],
+                ['text' => 'Second Checkbox', 'value' => 2]
+            ],
+            'optionsLabel' => ['class' => ['custom-class', 'another-class'], 'some-other-attr' => 'blah'],
+            'value' => ['1']
+
+        ]);
```

Comment:
This is a really long test method. Could it be broken up into smaller tests that are co-located with the method under test?